spring-grpc-test/src/main/java/org/springframework/grpc/test/InProcessGrpcServerFactoryAutoConfiguration.java

```diff
@@ -49,8 +52,14 @@ InProcessGrpcServerFactory grpcServerFactory(GrpcServiceDiscoverer grpcServicesD
 	}
 
 	@Bean
-	InProcessGrpcChannelFactory grpcChannelFactory() {
-		InProcessGrpcChannelFactory factory = new InProcessGrpcChannelFactory();
+	@ConditionalOnMissingBean
```

Comment:
I don't like that this bean is repeated here. If we want to pass one into the channel factory we need to create it here or split the client auto-config to include a `@Configuration` that both could import (this AC has a before condition on client AC).